serve property in webpack.config.js is treated as invalid

<!-- Please don't delete this template or we'll close your issue -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->

# Bug report

<!-- Please ask questions on StackOverflow or the webpack Gitter. -->
<!-- https://stackoverflow.com/questions/ask?tags=webpack -->
<!-- https://gitter.im/webpack/webpack -->
<!-- Issues which contain questions or support requests will be closed. -->

**What is the current behavior?**
A _webpack.config.js_ containing a `serve` property (to configure https://github.com/webpack-contrib/webpack-serve) is treated as invalid.


**If the current behavior is a bug, please provide the steps to reproduce.**
Use a _webpack.config.js_ which contains the `serve` property and start webpack.

<!-- A great way to do this is to provide your configuration via a GitHub repo. -->
<!-- Best provide a minimal reproduceable repo with instructions -->
<!-- Repos with too many files or long configs are not suitable -->
<!-- Please only add small snippets of code directly into the issue -->
<!-- https://gist.github.com is a good place for longer code snippets -->
<!-- If your issue is caused by a plugin or loader file the issue on the plugin/loader repo instead. -->

**What is the expected behavior?**
A _webpack.config.js_ containing a `serve` property should not be treated as invalid. The `devServer` property is also not treated as invalid.

<!-- "It should work" is not a good explaination -->
<!-- Explain how exactly you expecting it to behave -->

**Other relevant information:**
webpack version:  4.8.3
Node.js version: v10.1.0
Operating System: Linux 4.16.8
Additional tools: webpack-serve 1.0.2

```
npm run build

> danube@ build /home/user1/webapp
> webpack --mode production

Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration has an unknown property 'serve'. These properties are valid:
   object { mode?, amd?, bail?, cache?, context?, dependencies?, devServer?, devtool?, entry?, externals?, loader?, module?, name?, node?, output?, optimization?, parallelism?, performance?, plugins?, profile?, recordsInputPath?, recordsOutputPath?, recordsPath?, resolve?, resolveLoader?, stats?, target?, watch?, watchOptions? }
   For typos: please correct them.
   For loader options: webpack >= v2.0.0 no longer allows custom properties in configuration.
     Loaders should be updated to allow passing options via loader options in module.rules.
     Until loaders are updated one can use the LoaderOptionsPlugin to pass these options to the loader:
     plugins: [
       new webpack.LoaderOptionsPlugin({
         // test: /\.xxx$/, // may apply this only for some modules
         options: {
           serve: …
         }
       })
     ]
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! danube@ build: `webpack --mode production`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the danube@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user1/.npm/_logs/2018-05-15T10_14_28_664Z-debug.log
```

--- a/schemas/WebpackOptions.json
+++ b/schemas/WebpackOptions.json
@@ -1679,6 +1679,10 @@
         }
       ]
     },
+    "serve": {
+      "description": "Options for webpack-serve",
+      "type": "object"
+    },
     "stats": {
       "description": "Used by the webpack CLI program to pass stats options.",
       "anyOf": [

--- a/test/Validation.test.js
+++ b/test/Validation.test.js
@@ -199,7 +199,7 @@ describe("Validation", () => {
 				" - configuration has an unknown property 'postcss'. These properties are valid:",
 				"   object { mode?, amd?, bail?, cache?, context?, dependencies?, devServer?, devtool?, entry?, externals?, " +
 					"loader?, module?, name?, node?, output?, optimization?, parallelism?, performance?, plugins?, profile?, recordsInputPath?, " +
-					"recordsOutputPath?, recordsPath?, resolve?, resolveLoader?, stats?, target?, watch?, watchOptions? }",
+					"recordsOutputPath?, recordsPath?, resolve?, resolveLoader?, serve?, stats?, target?, watch?, watchOptions? }",
 				"   For typos: please correct them.",
 				"   For loader options: webpack >= v2.0.0 no longer allows custom properties in configuration.",
 				"     Loaders should be updated to allow passing options via loader options in module.rules.",

